Deprecated code in macOS Monterey

**Describe the bug**
There is a deprecation messages while building code with gopsutil v3.
```
# github.com/shirou/gopsutil/disk
iostat_darwin.c:28:2: warning: 'IOMasterPort' is deprecated: first deprecated in macOS 12.0 [-Wdeprecated-declarations]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/IOKit.framework/Headers/IOKitLib.h:132:1: note: 'IOMasterPort' has been explicitly marked deprecated here
# github.com/shirou/gopsutil/host
smc_darwin.c:75:41: warning: 'kIOMasterPortDefault' is deprecated: first deprecated in macOS 12.0 [-Wdeprecated-declarations]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/IOKit.framework/Headers/IOKitLib.h:123:19: note: 'kIOMasterPortDefault' has been explicitly marked deprecated here
```

**To Reproduce**
```go
package main

import (
	_ "github.com/shirou/gopsutil/cpu"
	_ "github.com/shirou/gopsutil/disk"
	_ "github.com/shirou/gopsutil/host"
	_ "github.com/shirou/gopsutil/mem"
)

func main() {}
```

**Expected behavior**
There isn't any messages about deprecation.

**Environment (please complete the following information):**
 - [x] Mac OS: [paste the result of `sw_vers` and `uname -a`
```
ProductName:	macOS
ProductVersion:	12.0.1
BuildVersion:	21A559
```
```
Darwin MacBook-Pro.local 21.1.0 Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64 x86_64
```

Hi, I have got the same warning in macOS 12.0.1, do you known how to fix it?

```
# github.com/shirou/gopsutil/cpu
../../../../pkg/mod/github.com/shirou/gopsutil@v2.20.7+incompatible/cpu/cpu_darwin_cgo.go:13:5: warning: 'TARGET_OS_MAC' is not defined, evaluates to 0 [-Wundef-prefix=TARGET_OS_]
# github.com/shirou/gopsutil/disk
iostat_darwin.c:28:2: warning: 'IOMasterPort' is deprecated: first deprecated in macOS 12.0 [-Wdeprecated-declarations]
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/IOKit.framework/Headers/IOKitLib.h:132:1: note: 'IOMasterPort' has been explicitly marked deprecated here
```

--- a/disk/iostat_darwin.c
+++ b/disk/iostat_darwin.c
@@ -25,7 +25,7 @@ gopsutil_v3_readdrivestat(DriveStats a[], int n)
 	kern_return_t status;
 	int na, rv;
 
-	IOMasterPort(bootstrap_port, &port);
+	IOMainPort(bootstrap_port, &port);
 	match = IOServiceMatching("IOMedia");
 	CFDictionaryAddValue(match, CFSTR(kIOMediaWholeKey), kCFBooleanTrue);
 	status = IOServiceGetMatchingServices(port, match, &drives);

--- a/disk/iostat_darwin.h
+++ b/disk/iostat_darwin.h
@@ -30,3 +30,7 @@ struct CPUStats {
 };
 
 extern int gopsutil_v3_readdrivestat(DriveStats a[], int n);
+
+#if (MAC_OS_X_VERSION_MIN_REQUIRED < 120000) // Before macOS 12 Monterey
+       #define IOMainPort IOMasterPort
+#endif

--- a/host/smc_darwin.c
+++ b/host/smc_darwin.c
@@ -72,7 +72,7 @@ kern_return_t gopsutil_v3_open_smc(void) {
   kern_return_t result;
   io_service_t service;
 
-  service = IOServiceGetMatchingService(kIOMasterPortDefault,
+  service = IOServiceGetMatchingService(kIOMainPortDefault,
                                         IOServiceMatching(IOSERVICE_SMC));
   if (service == 0) {
     // Note: IOServiceMatching documents 0 on failure

--- a/host/smc_darwin.h
+++ b/host/smc_darwin.h
@@ -29,4 +29,9 @@ kern_return_t gopsutil_v3_open_smc(void);
 kern_return_t gopsutil_v3_close_smc(void);
 double gopsutil_v3_get_temperature(char *);
 
+#if (MAC_OS_X_VERSION_MIN_REQUIRED < 120000) // Before macOS 12 Monterey
+	#define kIOMainPortDefault kIOMasterPortDefault
+#endif
+
+
 #endif // __SMC_H__

